### DISCUSSION TITLE
Approaches to add `min` and `max` to a `numberField`

### DISCUSSION THREAD

**Discussion by amitaibu** [digitallyinduced/ihp#1204] [General]
Currently, all input fields share the same [FormField](https://github.com/digitallyinduced/ihp/blob/1998c197228e3fe87bc507a562456bb1e7f7ca28/IHP/View/Types.hs#L45-L64).

If we'd like for example to have `<input type="number" min="0" ... />` then we need to opt-out to writing HSX, but it requires a lot of manual work. Roughly like this:

```
renderForm :: Post -> Html
renderForm post = [hsx|
    <div class="form-group">
        <label>
            Number field
        </label>
        <input type="number" name="someval" value={get #someval post} min=0/>
        {somevalFeedback}
    </div>
|]
    where
        isInvalidSomeval = isJust (getValidationFailure #someval post)

        somevalFeedback = case getValidationFailure #someval post of
            Just result -> [hsx|<div class="invalid-feedback">{result}</div>|]
            Nothing -> mempty
```

Unless I'm getting things wrong, I think we have a few options:

1. Add an `attributes :: [Attributes]` so one could add any attribute they would want. A disadvantage would be that one could add `min` to a non-number field.
2. Break `FormField` to smaller pieces, as that's already a requirement of https://github.com/digitallyinduced/ihp/pull/1004
3. Tell people to opt to the hsx version, but have helper function(s) to lower the verbosity of the code needed.

**Comment by mpscholten** [comment]
On 1: Actually right now you can already do this by following this guide: https://ihp.digitallyinduced.com/Guide/form.html#adding-custom-attributes-to-the-input-element

Longterm I see that 2 is the way forward 👍
